### PR TITLE
fix: no asar support (#4), renamed directories (#5) and check if we need to duplicate asar's (#2)

### PR DESCRIPTION
**Pull request by NielsLeenheer**
Fixes all issues from #4, #5 and #2.

The fix for #4 was pretty simple. Just prepare two separate entry-asar files. One for asar's and one in case there is no asar. And copy the right one for each situation.

The fix for #5 was renaming the directories to `app-x64` and `app-arm64` instead of `x64.app` and `arm64.app`. I've also renamed the asar files, to match this new naming scheme. 

For #2, I now check if the asar or app directories are identical. If they are, then we can leave the exiting asar or app directory just in place in the tmp directory. No need to rename and create and entry-asar. In case of an asar, I use the already existing sha function to check if they are identical. In case of an app directory I use a new dependency `dir-compare` to check the directories by size and content.

### CODE REVIEW

File: entry-asar/has-asar.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+if (process.arch === 'arm64') {
+  process._archPath = require.resolve('../app-arm64.asar');
+} else {
+  process._archPath = require.resolve('../app-x64.asar');
+}
+
+require(process._archPath);
```
Comment (MarshallOfSound) [inline entry-asar/has-asar.js:7]: `prettier` should have done this for us but can you add the trailing new line in here?

File: entry-asar/no-asar.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+if (process.arch === 'arm64') {
+  process._archPath = require.resolve('../app-arm64');
+} else {
+  process._archPath = require.resolve('../app-x64');
+}
+
+require(process._archPath);
```
Comment (MarshallOfSound) [inline entry-asar/no-asar.js:7]: Same here, missing new line

**Review by MarshallOfSound — COMMENTED** [review]
This all seems legit enough 😄

**Comment by NielsLeenheer** [comment]
Fixed! Added a new line to both entry-asar files.

**Review by MarshallOfSound — APPROVED** [review]

**Pull request merged**